example/moving_average.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2023 Jiri Nytra (jiri.nytra at gmail dot com)
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <flux.hpp>
+
+#include <cstddef>
+#include <cassert>
+#include <deque>
+#include <vector>
+
+struct sliding_window_t
+{
+    explicit sliding_window_t(std::size_t N): size(N) {
+    }
+
+    void push(int value) {
+        if (window.size() < size) {
+            sum += value;
+        } else {
+            sum += value - window.front();
+            window.pop_front();
+        }
+
+        window.push_back(value);
+    }
+
+    int average() const {
+        return sum / window.size();
+    }
+
+    std::size_t size;
+    std::deque<int> window;
+    int sum = 0;
+};
+
+auto sliding_window = [](sliding_window_t&& win, int next) {
+    win.push(next);
+    return std::move(win);
+};
+
+int main() {
+    std::vector intervals = {1, 5, 6, 1, 2, 9, 7, -1, 0};
+
+    // compute moving average by scan adaptor (more effective for large windows)
+    auto ma = flux::from(intervals)
+        .scan(sliding_window, sliding_window_t(3))
+        .map(flux::proj(&sliding_window_t::average))
```

Comment:
I think you should be able to pass the mem fn ptr directly to `map`, without needing to use `proj`:

```suggestion
        .map(&sliding_window_t::average)
```